### PR TITLE
Encode UUIDs with Base 64

### PR DESCRIPTION
**Pull request by odisseus**

### CODE REVIEW

**Comment by propensive** [comment]
What's the motivation for this, @odisseus?

**Comment by jvican** [comment]
@propensive The motivation is shorter names, which I kind of like too https://github.com/scalacenter/bloop/issues/848

File: backend/src/main/scala/bloop/util/Base64Uuid.scala [hunk 1]
```diff
@@ -0,0 +1,27 @@
+package bloop.util
+import java.nio.ByteBuffer
+import java.util.{Base64, UUID}
+
+object Base64Uuid {
```
Comment (jvican) [inline backend/src/main/scala/bloop/util/Base64Uuid.scala]: Let's rename to `UUIDUtil`

File: backend/src/main/scala/bloop/util/Base64Uuid.scala [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+  // Copied from https://stackoverflow.com/a/29836273
+  implicit class UuidWithBase64(val uuid: UUID) extends AnyVal{
```
Comment (jvican) [inline backend/src/main/scala/bloop/util/Base64Uuid.scala]: I propose to turn this into a method, there's no need for a value class

File: backend/src/main/scala/bloop/util/Base64Uuid.scala [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    def toBase64String: String = {
+      val bb = ByteBuffer.wrap(new Array[Byte](16))
+      bb.putLong(uuid.getMostSignificantBits)
+      bb.putLong(uuid.getLeastSignificantBits)
+      Base64.getUrlEncoder.encodeToString(bb.array)
+    }
+  }
+
+  def apply(base64: String): UUID = {
```
Comment (jvican) [inline backend/src/main/scala/bloop/util/Base64Uuid.scala]: Is this being used? If not, let's remove it, we should be never decoding Base 64 to UUID, all we want is a shorter UUID for a random id :smile:

File: backend/src/main/scala/bloop/util/Base64Uuid.scala [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    val bytes = Base64.getUrlDecoder.decode(base64)
+    val bb = ByteBuffer.wrap(bytes)
+    val high = bb.getLong
+    val low = bb.getLong
+    new UUID(high, low)
+  }
+
+  def randomUUID(): String = UUID.randomUUID.toBase64String
```
Comment (jvican) [inline backend/src/main/scala/bloop/util/Base64Uuid.scala]: `def randomUUID: String` without `()` is more idiomatic

**Comment by propensive** [comment]
Yes, I like it too. Base64 will use two or three characters in addition to the alphanumeric characters. These will be `+`, `/` and potentially `=` at the end, which aren't filesystem-safe. I didn't check what this version produces, but there should be a URL-safe version of Base64 which uses `-` and `_` instead.

**Comment by jvican** [comment]
Good point @propensive , even though @odisseus linked to https://en.wikipedia.org/wiki/Base64#Filenames in his ticket, it doesn't look like this PR is using the filename safe base64 variant.

**Comment by odisseus** [comment]
The method [`getUrlEncoder`](https://docs.oracle.com/javase/8/docs/api/java/util/Base64.html#getUrlEncoder--) of class `java.util.Base64` promises to return an encoder that produces values [safe](https://docs.oracle.com/javase/8/docs/api/java/util/Base64.html#url) for use in filenames and URLs, but I have to admit I haven't verified that.

**Comment by propensive** [comment]
```
scala> java.util.Base64.getUrlEncoder.encodeToString((0 to 255).to[Array].map(_.toByte))
res1: String = AAECAwQFBgcICQoLDA0ODxAREhMUFRYXGBkaGxwdHh8gISIjJCUmJygpKissLS4vMDEyMzQ1Njc4OTo7PD0-P0BBQkNERUZHSElKS0xNTk9QUVJTVFVWV1hZWltcXV5fYGFiY2RlZmdoaWprbG1ub3BxcnN0dXZ3eHl6e3x9fn-AgYKDhIWGh4iJiouMjY6PkJGSk5SVlpeYmZqbnJ2en6ChoqOkpaanqKmqq6ytrq-wsbKztLW2t7i5uru8vb6_wMHCw8TFxsfIycrLzM3Oz9DR0tPU1dbX2Nna29zd3t_g4eLj5OXm5-jp6uvs7e7v8PHy8_T19vf4-fr7_P3-_w==
```

Looks good to me.

File: backend/src/main/scala/bloop/util/Base64Uuid.scala (hunk 4 above)
Comment (odisseus) [reply backend/src/main/scala/bloop/util/Base64Uuid.scala]: Ok

File: backend/src/main/scala/bloop/util/Base64Uuid.scala (hunk 3 above)
Comment (odisseus) [reply backend/src/main/scala/bloop/util/Base64Uuid.scala]: Removed

File: backend/src/main/scala/bloop/util/Base64Uuid.scala (hunk 2 above)
Comment (odisseus) [reply backend/src/main/scala/bloop/util/Base64Uuid.scala]: Done

File: backend/src/main/scala/bloop/util/Base64Uuid.scala (hunk 1 above)
Comment (odisseus) [reply backend/src/main/scala/bloop/util/Base64Uuid.scala]: Done

**Comment by odisseus** [comment]
@jvican, can you please check why the build has failed? The logs suggest it's an infrastructure issue.

**Comment by jvican** [comment]
There was an outage at EPFL that brought our Windows CI down. Whenever it's back up, I'll merge this.

**Pull request merged**